components/renku_data_services/notebooks/api/amalthea_patches/init_containers.py

```diff
@@ -105,7 +111,8 @@ async def git_clone_container_v2(server: "UserServer") -> dict[str, Any] | None:
         )
 
     # Set up git providers
-    required_git_providers = await server.required_git_providers()
+    required_provider_ids: set[str] = set(r.provider for r in repositories if r.provider)
```

Comment:
This can be written as a set literal as well: `{r.provider for r in repositories if r.provider}`